File: SDKs/squareup.com/2.0/python/src/sdk/models/operations/createlocation.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class CreateLocationSecurity:
    oauth2: shared.SchemeOauth2 = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclass
class CreateLocationRequest:
    request: shared.CreateLocationRequest = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    security: CreateLocationSecurity = field(default=None)
    

@dataclass
class CreateLocationResponse:
    content_type: str = field(default=None)
    create_location_response: Optional[shared.CreateLocationResponse] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/squareup.com/2.0/typescript/src/sdk/models/operations/createlocation.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class CreateLocationSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;
}


export class CreateLocationRequest extends SpeakeasyBase {
  @Metadata({ data: "request, media_type=application/json" })
  request: shared.CreateLocationRequest;

  @Metadata()
  security: CreateLocationSecurity;
}


export class CreateLocationResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  createLocationResponse?: shared.CreateLocationResponse;

  @Metadata()
  statusCode: number;
}
